src/Communicate/Buffers.hpp

```diff
@@ -21,21 +21,20 @@
 //
 
 namespace ippl {
+    namespace mpi {
 
-    template <typename MemorySpace, typename T>
-    Communicate::buffer_type<MemorySpace> Communicate::getBuffer(int id, size_type size,
-                                                                 double overallocation) {
-        auto& buffers = buffers_m.get<MemorySpace>();
-        size *= sizeof(T);
-        if (buffers.contains(id)) {
-            if (buffers[id]->getBufferSize() < size) {
+        template <typename MemorySpace, typename T>
+        Communicator::buffer_type<MemorySpace> Communicator::getBuffer(int id, size_type size,
+                                                                       double overallocation) {
+            auto& buffers = buffers_m.get<MemorySpace>();
+            size *= sizeof(T);
+            if (buffers.contains(id) && buffers[id]->getBufferSize() < size) {
                 buffers[id]->reallocBuffer(size);
             }
+            buffers[id] = std::make_shared<archive_type<MemorySpace>>(
```

Comment:
I am a bit confused here.. This to me seems will always try to allocate the buffer with id even if we have enough size which we want to avoid because of performance in GPU. Previously we do this only if the id doesn't exist, or did I misunderstand something here?

Reply:
This is a mistake caused by a merge conflict. The buffer should be returned after it's reallocated when the buffer already exists.

Reply:
No I think it was one of my commits that lost the `return` and I just didn't notice at the time.

Reply:
Ok.

Reply:
It still seems to me that this problem was not fixed. Or am I wrong?